src/PerfCounters.cc

```diff
@@ -860,6 +582,22 @@ Ticks PerfCounters::read_ticks(Task* t) {
     }
   }
 
+  if (fd_strex_counter.is_open()) {
+    uint64_t strex_count = read_counter(fd_strex_counter);
+    if (strex_count > 0) {
+      LOG(debug) << strex_count << " strex detected";
+      if (!Flags::get().force_things) {
+        ASSERT(t, false)
+            << strex_count
+            << " (speculatively) executed strex instructions detected. \n"
+               "On aarch64, rr only supports applications making use of LSE\n"
+               "atomics rather than legacy LL/SC-based atomics.\n"
+               "Aborting. Retry with -F to override, but it will probably\n"
+               "fail.";
```

Comment:
make clear that it's *replay* that will probably fail.

Reply:
Good point.